lighthouse-core/gather/gather-runner.js

```diff
@@ -259,9 +259,14 @@ class GatherRunner {
       mainRecord = NetworkAnalyzer.findMainDocument(networkRecords, passContext.url);
     } catch (_) {}
 
+    let finalRecord;
+    try {
+      finalRecord = NetworkAnalyzer.findFinalDocument(mainRecord);
+    } catch (_) {}
+
```

Comment:
I don't think we need `try/catch` around this if we end up going with the `resolveRedirects` option that can handle `undefined` :)

```suggestion
    // MIME Type is only set on the final redirected document request. Use this for the HTML check instead of root.
    const finalRecord = NetworkAnalyzer.resolveRedirects(mainRecord);

```